packages/corelib/src/playout/processAndPrune.ts

```diff
@@ -10,12 +10,49 @@ import { ReadonlyDeep } from 'type-fest'
 /**
  * Get the `enable: { start: ?? }` for the new piece in terms that can be used as an `end` for another object
  */
-function getPieceStartTimeAsReference(newPieceStart: number | 'now'): number | RelativeResolvedEndCap {
-	return typeof newPieceStart === 'number' ? newPieceStart : { offsetFromNow: 0 }
+function getPieceStartTimeAsReference(
+	newPieceStart: number | 'now',
+	partTimes: PartCurrentTimes,
+	pieceToAffect: ReadonlyDeep<PieceInstance>
+): number | RelativeResolvedEndCap {
+	if (typeof newPieceStart !== 'number') return { offsetFromNow: 0 }
+
+	if (pieceToAffect.piece.enable.isAbsolute) {
+		// If the piece is absolute timed, then the end needs to be adjusted to be absolute
+		if (pieceToAffect.piece.enable.start === 'now') {
+			return { offsetFromNow: newPieceStart }
+		} else {
+			// Translate to an absolute timestamp
+			return partTimes.currentTime - partTimes.nowInPart + newPieceStart
+		}
+	}
+
+	return newPieceStart
 }
 
-function getPieceStartTimeWithinPart(p: ReadonlyDeep<PieceInstance>): 'now' | number {
-	return p.piece.enable.start
+function getPieceStartTimeWithinPart(p: ReadonlyDeep<PieceInstance>, partTimes: PartCurrentTimes): 'now' | number {
+	const pieceEnable = p.piece.enable
+	if (pieceEnable.isAbsolute) {
+		// Note: these can't be adlibbed, so we don't need to consider adding the preroll
+
+		if (pieceEnable.start === 'now') {
+			// Should never happen, but just in case
+			return pieceEnable.start
+		} else {
+			// Translate this to the part
+			return pieceEnable.start - partTimes.currentTime + partTimes.nowInPart
+		}
+	}
+
+	// If the piece is dynamically inserted, then its preroll should be factored into its start time, but not for any infinite continuations
+	const isStartOfAdlib =
+		!!p.dynamicallyInserted && !(p.infinite?.fromPreviousPart || p.infinite?.fromPreviousPlayhead)
+
+	if (isStartOfAdlib && pieceEnable.start !== 'now') {
+		return pieceEnable.start + (p.piece.prerollDuration ?? 0)
+	} else {
+		return pieceEnable.start
+	}
```

Comment:
I am a bit concerned about this block, as this was recently changed by NRK as it was causing timing issues  https://github.com/Sofie-Automation/sofie-core/pull/1441

I think this change will be the cause of the currently failing test `Current part with preroll and adlib preroll`